panda/src/display/graphicsStateGuardian.cxx

```diff
@@ -230,6 +230,7 @@ GraphicsStateGuardian(CoordinateSystem internal_coordinate_system,
   _supports_generate_mipmap = false;
   _supports_depth_texture = false;
   _supports_depth_stencil = false;
+  _supports_luminance_texture = false;
```

Comment:
So setting "gl-version 3 2" in Config.prc isn't sufficient to get a core-only context in Mesa?  If I recall correctly, the spec guarantees that we get a core-only context by default when requesting "gl-version 3 2" or above.

My biggest concern is that our context creation logic is already very complex.  This PR seems to add a whole lot more logic to it to the point where it's really hard to follow what it does and how it's supposed to work.

You could consider splitting out the context stuff from this pull request to make it easier to review separately.  I certainly have no objections with getting the GSG changes in.

As for D3D9, we are probably safe to assume it has luminance support.